service/dap/server.go

```diff
@@ -70,9 +71,10 @@ type Server struct {
 	// args tracks special settings for handling debug session requests.
 	args launchAttachArgs
 
+	mu sync.Mutex
```

Comment:
Let's call this noDebugProcessMux since we might end up with more locks and will need to differentiate.

Reply:
Based no the comment in the other PR, I understand now what your intension was here. Please disregard this comment.